TwigExtension.php

```diff
@@ -29,17 +29,61 @@ public function initErrorLogger($level = 'error', $includeScriptTag = true)
 (function () {
     var oldErrorHandler = window.onerror;
 
-    window.onerror = function(errorMsg, file, line) {
+    window.onerror = function(errorMsg, file, line, col, error) {
         var key,
             e = encodeURIComponent,
             customContext = window.nelmio_js_logger_custom_context,
             customContextStr = '';
 
         if (oldErrorHandler) {
-            oldErrorHandler(errorMsg, file, line);
+            oldErrorHandler(errorMsg, file, line, col, error);
         }
 
-        if ('object' === typeof customContext) {
+        if( typeof StackTrace !== 'undefined' && typeof StackTrace.fromError === 'function'){
+            window.logWithStacktrace(errorMsg, file, line, col, error);
+        } else {
+            window.logWithoutStacktrace(errorMsg, file, line, col, error);
+        }
+    };
+
+    window.logWithStacktrace = function(errorMsg, file, line, col, error){
```

Comment:
should these functions really be leaked as global variables ? this seems really wrong to me